docs/tutorials/flow-task-config.md

```diff
@@ -94,6 +94,26 @@ def my_flow(name: str, date: datetime.datetime):
 my_flow(name="marvin", date=datetime.datetime.utcnow())
 ```
 
+Additionally this setting also accepts a function that returns a string for the flow_run_name
+
+```python
+import datetime
+from prefect import flow
+
+def generate_flow_run_name():
+    date = datetime.datetime.utcnow()
+
+    return f"{date:%A}-is-a-nice-day"
+
+@flow(flow_run_name=generate_flow_run_name)
+def my_flow(name: str):
+    pass
+
+# creates a flow run called 'Thursday-is-a-nice-day'
+my_flow(name="marvin")
+```
+
```

Comment:
Can we add a blurb here that says something like

> If you need access to information about the flow, use the `prefect.runtime` module. For example:

Then include an example that pulls the name of the flow and a parameter from the runtime?

Reply:
done